app/src/main/java/com/jvoye/tasky/auth/presentation/register/RegisterViewModel.kt

```diff
@@ -1,62 +1,76 @@
 package com.jvoye.tasky.auth.presentation.register
 
 import androidx.compose.foundation.text.input.TextFieldState
-import androidx.compose.runtime.getValue
-import androidx.compose.runtime.mutableStateOf
-import androidx.compose.runtime.setValue
 import androidx.compose.runtime.snapshotFlow
 import androidx.lifecycle.ViewModel
 import androidx.lifecycle.viewModelScope
 import com.jvoye.tasky.R
 import com.jvoye.tasky.auth.domain.UserDataValidator
 import com.jvoye.tasky.core.presentation.designsystem.util.UiText
+import kotlinx.coroutines.flow.MutableStateFlow
+import kotlinx.coroutines.flow.SharingStarted
 import kotlinx.coroutines.flow.launchIn
 import kotlinx.coroutines.flow.onEach
+import kotlinx.coroutines.flow.onStart
+import kotlinx.coroutines.flow.stateIn
+import kotlinx.coroutines.flow.update
 
 class RegisterViewModel(
     private val userDataValidator: UserDataValidator
 ): ViewModel() {
+    private val _state = MutableStateFlow(RegisterState())
+    val state = _state
+        .onStart { validateUserInput() }
```

Comment:
It typically makes sense to add some kind of flag to check whether the initial data has already been loaded. Right now, this code would be called with every resubscription of the Flow (e.g. when rotating the device, navigating back to this screen or opening the app from the background).
This would currently create a new independent Flow subscriber for the text to validate it. 

Something like this fixes it:
```kotlin
private var hasLoadedInitialData = false

...
onStart {
    if(!hasLoadedInitialData) {
        validateUserInput()
        hasLoadedInitialData = true
    }
}
```